src/controller.coffee

```diff
@@ -1948,6 +1968,15 @@ define ['droplet-helper',
 
     socket.notifyChange()
 
+  Editor::populateBlock = (block, string) ->
+    newBlock = @mode.parse(string, wrapAtRoot: false).start.next.container
+
+    if newBlock?.start?.next?
+      block.start.append newBlock.start.next
+      newBlock.end.prev.append block.end
+      block.notifyChange()
```

Comment:
It's interesting that you replace the contents of the block but do not replace the block itself, which is counterintuitive.  Was there a reason that you needed to do that?  Also, I'd love Anthony to weigh in on whether anything is missing here.